seed/seed.go

```diff
@@ -235,13 +235,13 @@ func ReadSystemEssential(seedDir, label string, essentialTypes []snap.Type, tm t
 // a earliestTime lower bound for current time.
 // It returns as well an improved lower bound by considering
 // appropriate assertions in the seed.
-func ReadSystemEssentialAndBetterEarliestTime(seedDir, label string, essentialTypes []snap.Type, earliestTime time.Time, numJobs int, tm timings.Measurer) (*asserts.Model, []*Snap, time.Time, error) {
+func ReadSystemEssentialAndBetterEarliestTime(seedDir, label string, essentialTypes []snap.Type, earliestTime time.Time, numJobs int, tm timings.Measurer) (*asserts.Model, []*Snap, *asserts.Database, time.Time, error) {
```

Comment:
there should be a way to have this return snap-revisions as well directly

Reply:
You mean in addition to returning the entire db or instead of it? 

I needed the entire db here because I use it later in [cmd/snap-bootstrap/initramfs_verity.go](https://github.com/snapcore/snapd/pull/12664/commits/a41b0406284f0d467b77ed67fcc16765112d2641#diff-cb177debf21ce86e6a5a24af7ebe6051065b126c882b4c0aeb9421cf4c368616R38) to go from snap names to snap revisions:
1) query snap-declarations by name to retrieve snap-ids
2) query snap-revisions using the snap-ids

Is there a better way to do this?